src/lib/components/explorer/results/ResultsPanel.svelte

```diff
@@ -100,51 +44,53 @@
   );
 
   let showExplorerDistributions = $derived(
-    !isOpenAccess && features.explorer.distributionExplorer && hasValidDistributionFilters,
+    isExplorer && features.explorer.distributionExplorer && hasValidDistributionFilters,
   );
 
   let showDiscoverDistributions = $derived(
     isOpenAccess && features.discoverFeautures.distributionExplorer && hasValidDistributionFilters,
   );
 
   let showVariantExplorer = $derived(
-    !isOpenAccess && features.explorer.variantExplorer && $hasGenomicFilter,
+    isExplorer && features.explorer.variantExplorer && $hasGenomicFilter,
+  );
+
+  let showCohortDetails = $derived(
+    isExplorer && features.explorer.enableCohortDetails && $hasNonZeroResult,
   );
 
   let showToolSuite = $derived(
-    totalPatients !== 0 &&
-      ($filters.length !== 0 || $exports.length !== 0) &&
-      (showExplorerDistributions || showDiscoverDistributions || showVariantExplorer),
+    ($filters.length !== 0 || $exports.length !== 0) &&
+      (showExplorerDistributions ||
+        showDiscoverDistributions ||
+        showVariantExplorer ||
+        showCohortDetails) &&
+      $hasNonZeroResult,
   );
 
-  onMount(async () => {
-    unsubFilters = filters.subscribe(() => {
-      triggerRefreshCount = getCount();
-    });
-  });
+  function subscribe() {
+    if (!unsubFilters) {
+      unsubFilters = filters.subscribe(() => loadPatientCount(isOpenAccess));
+    }
+  }
 
-  afterNavigate(async () => {
+  function unsubscribe() {
     if (unsubFilters) {
       unsubFilters();
+      unsubFilters = null;
     }
+  }
 
-    isOpenAccess = page.url.pathname.includes('/discover');
-    const isExplorer = page.url.pathname.includes('/explorer');
-    if (isExplorer || isOpenAccess) {
-      await tick();
-      unsubFilters = filters.subscribe(() => {
-        triggerRefreshCount = getCount();
-      });
+  $effect(() => {
+    if (currentPage !== page.url.pathname) {
+      currentPage = page.url.pathname;
+      unsubscribe();
+      subscribe();
```

Comment:
I would put a comment here as to why this is happening

Reply:
I add a pretty good comment in the cohort detail page, when I change the logic.

Reply:
Actually this is a great time to deploy this to BDC dev and ensure these changes don't regress the bug they were introduced for. I'll add a comment on why we have to do this whole effect block. My comment in the cohort detail PR does not relate to why we have to do this, which is what I think you're getting at.